homeassistant/components/remote_calendar/client.py

```diff
@@ -1,12 +1,74 @@
 """Specifies the parameter for the httpx download."""
 
-from httpx import AsyncClient, Response, Timeout
+import logging
+from urllib.parse import parse_qs, urlparse
 
+from httpx import AsyncClient, Auth, BasicAuth, Response, Timeout
 
-async def get_calendar(client: AsyncClient, url: str) -> Response:
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_calendar(
+    client: AsyncClient,
+    url: str,
+    username: str | None = None,
+    password: str | None = None,
+) -> Response:
     """Make an HTTP GET request using Home Assistant's async HTTPX client with timeout."""
-    return await client.get(
+    headers: dict[str, str] = {}
+    auth: Auth | None = None
+
+    # Set up HTTP Basic Auth if credentials provided
+    if username and password:
+        auth = BasicAuth(username, password)
+        _LOGGER.debug("HTTP Basic Auth enabled for username: %s", username)
+
+    # Extract API key from URL query parameters and add as header
+    # Some services (like Radarr) prefer header-based authentication
```

Comment:
Not sure i follow the rationale for this block. Taking a query parameter and also adding a header? Why is it a query parameter if the service prefers it be a header? This does not seem appropriate.